x-pack/plugins/dataset_quality/public/hooks/use_dataset_quality_table.tsx

```diff
@@ -28,63 +28,83 @@ export const useDatasetQualityTable = () => {
   const {
     services: { fieldFormats },
   } = useKibanaContextForPlugin();
-  const [selectedDataset, setSelectedDataset] = useState<DataStreamStat>();
-  const [pageIndex, setPageIndex] = useState(0);
-  const [pageSize, setPageSize] = useState(10);
-  const [sortField, setSortField] = useState<SORT_FIELD>(DEFAULT_SORT_FIELD);
-  const [sortDirection, setSortDirection] = useState<DIRECTION>(DEFAULT_SORT_DIRECTION);
-
-  const defaultTimeRange = getDefaultTimeRange();
-
-  const { dataStreamsStatsServiceClient: client } = useDatasetQualityContext();
-  const { data = [], loading } = useFetcher(async () => client.getDataStreamsStats(), []);
-  const { data: degradedStats = [], loading: loadingDegradedStats } = useFetcher(
-    async () =>
-      client.getDataStreamsDegradedStats({
-        start: defaultTimeRange.from,
-        end: defaultTimeRange.to,
-      }),
-    []
+
+  const { service } = useDatasetQualityContext();
+
+  const { page, rowsPerPage, sort } = useSelector(service, (state) => state.context.table);
+
+  const flyout = useSelector(service, (state) => state.context?.flyout);
+
+  const dataStreamStats = useSelector(service, (state) => state.context.dataStreamStats);
+  const loading = useSelector(service, (state) => state.matches('datasets.fetching'));
+
+  const degradedStats = useSelector(service, (state) => state.context.degradedDocStats);
+  const loadingDegradedStats = useSelector(service, (state) =>
+    state.matches('degradedDocs.fetching')
+  );
+
+  const flyoutClosed = useSelector(service, (state) => state.matches('datasets.loaded.idle'));
+
+  const closeFlyout = useCallback(() => service.send({ type: 'CLOSE_FLYOUT' }), [service]);
+  const openFlyout = useCallback(
+    (selectedDataset: DataStreamStat) => {
+      if (flyout?.dataset?.rawName === selectedDataset.rawName) {
+        service.send({
+          type: 'CLOSE_FLYOUT',
+        });
+
+        return;
+      }
+
+      service.send({
+        type: flyoutClosed ? 'OPEN_FLYOUT' : 'SELECT_NEW_DATASET',
+        dataset: selectedDataset as FlyoutDataset,
```

Comment:
**nit**: no need for a type assertion here, you can annotate the function argument as `FlyoutDataset`.

Reply:
Addressed in [2224843](https://github.com/elastic/kibana/pull/174906/commits/22248430df33bd43a5caffea88a35c9f8af7b1d6).